test/ebpf/testSockProg.c

```diff
@@ -0,0 +1,428 @@
+    // Map created. Now pin the map.
+    int error = bpf_obj_pin(fd, pin_path);
+    if (error != 0)
+    {
+        // close map fd.
+        _close(fd);
+        return -1;
+    }
+    return 0;
+}
+
+struct npm_endpoint_prog_t test_ebpf_prog()
+{
+    const char *connect4_program_name = "authorize_connect4";
+    const char *connect6_program_name = "authorize_connect6";
+    const char *recv4_accept_program_name = "authorize_recv_accept4";
+    const char *recv6_accept_program_name = "authorize_recv_accept6";
+    struct npm_endpoint_prog_t _npm_endpoint_prog_t;
+    struct bpf_object *object;
+    int program_fd;
+    int result = bpf_prog_load("src/endpoint_prog.o", BPF_PROG_TYPE_CGROUP_SOCK_ADDR, &object, &program_fd);
+
+    if (!object)
+    {
+        printf("object is null\n");
+        return _npm_endpoint_prog_t;
+    }
+    printf("Loaded program\n");
+    printf("%d program fd\n", program_fd);
+    printf("%d result\n", result);
+
+    if (result < 0)
+    {
+        printf("Load program failed\n");
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.object = object;
+
+    printf("Getting the bpf_prog for connect program\n");
+    struct bpf_program *connect_program = bpf_object__find_program_by_name(object, connect4_program_name);
+    if (!connect_program)
+    {
+        printf("%s is null\n", connect4_program_name);
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.connect4_program = connect_program;
+    printf("connect program\n");
+
+    struct bpf_program *connect6_program = bpf_object__find_program_by_name(object, connect6_program_name);
+    if (!connect6_program)
+    {
+        printf("%s is null\n", connect6_program_name);
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.connect6_program = connect6_program;
+    printf("connect6 program\n");
+
+    struct bpf_program *recv_accept_program = bpf_object__find_program_by_name(object, recv4_accept_program_name);
+    if (!recv_accept_program)
+    {
+        printf("%s is null\n", recv4_accept_program_name);
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.recv4_accept_program = recv_accept_program;
+    printf("recv program\n");
+
+    struct bpf_program *recv6_accept_program = bpf_object__find_program_by_name(object, recv6_accept_program_name);
+    if (!recv6_accept_program)
+    {
+        printf("%s is null\n", recv6_accept_program_name);
+        return _npm_endpoint_prog_t;
+    }
+
+    _npm_endpoint_prog_t.recv4_accept_program = recv6_accept_program;
+    printf("recv6 program\n");
+    printf("Now getting MAP fds and pinning them\n");
+
+    struct bpf_map *map_policy_maps_obj = bpf_object__find_map_by_name(object, "map_policy_maps");
+    if (map_policy_maps_obj == NULL)
+    {
+        printf("global policy map is null\n");
+        return _npm_endpoint_prog_t;
+    }
+
+    int err = pin_given_map(GLOBAL_POLICY_MAP, bpf_map__fd(map_policy_maps_obj));
```

Comment:
if you just want to pin all maps in the object, you can just call [bpf_object__pin_maps()](https://microsoft.github.io/ebpf-for-windows/libbpf_8h.html#ae1bffc48512a68600518c0f5d516b87e)

Reply:
Awesome will try this.